### PR TITLE
Modify the number of hugepages allocated

### PR DESCRIPTION
**Pull request by yafu-1**
Signed-off-by: Yan Fu <yafu@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virtual_device/filesystem_device.py [hunk 1]
```diff
@@ -141,16 +141,16 @@ def shared_data(vm_names, fs_devs):
             vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_names[index])
             vmxml_backup = vmxml.copy()
             vmxml_backups.append(vmxml_backup)
-            huge_pages_num += 1024
+            if int(vmxml.max_mem) < int(2048000):
+                vmxml.max_mem = int(2048000)
+            huge_pages_num = vmxml.max_mem // 2048 + 128
```
[the remark this answers is not in the document]
Comment (sathnaga) [reply libvirt/tests/src/virtual_device/filesystem_device.py:146]: @dzhengfy yep, good to have generic support for all arch.

@yafu-1  you could re-use this, if test needs only default hugepage size supported, https://github.com/avocado-framework/avocado-vt/blob/master/virttest/test_setup.py#L468

this library adds has lot other functions for hugepage setup in a system in a generic way, checkout if you could use them in here....

[the remark this answers is not in the document]
Comment (yafu-1) [reply libvirt/tests/src/virtual_device/filesystem_device.py]: Thanks for your suggestion. Updated.

File: libvirt/tests/src/virtual_device/filesystem_device.py [hunk 2]
```diff
@@ -141,16 +141,16 @@ def shared_data(vm_names, fs_devs):
             vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_names[index])
             vmxml_backup = vmxml.copy()
             vmxml_backups.append(vmxml_backup)
-            huge_pages_num += 1024
+            if int(vmxml.max_mem) < int(2048000):
```
[the remark this answers is not in the document]
Comment (yafu-1) [reply libvirt/tests/src/virtual_device/filesystem_device.py]: Updated.

File: libvirt/tests/src/virtual_device/filesystem_device.py [hunk 3]
```diff
@@ -141,16 +141,16 @@ def shared_data(vm_names, fs_devs):
...
+                vmxml.max_mem = int(2048000)
```
[the remark this answers is not in the document]
Comment (yafu-1) [reply libvirt/tests/src/virtual_device/filesystem_device.py]: Updated.

File: libvirt/tests/src/virtual_device/filesystem_device.py [hunk 4]
```diff
@@ -141,16 +142,18 @@ def shared_data(vm_names, fs_devs):
             vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_names[index])
             vmxml_backup = vmxml.copy()
             vmxml_backups.append(vmxml_backup)
-            huge_pages_num += 1024
+            if int(vmxml.max_mem) < 2048000:
+                vmxml.max_mem = 2048000
```
Comment (dzhengfy) [inline libvirt/tests/src/virtual_device/filesystem_device.py:146]: Is 2G required minimum value? If not, please help adjust it.

Comment (yafu-1) [reply libvirt/tests/src/virtual_device/filesystem_device.py]: I have adjust the memory to 1G.

File: libvirt/tests/src/virtual_device/filesystem_device.py [hunk 5]
```diff
@@ -141,16 +142,18 @@ def shared_data(vm_names, fs_devs):
             vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_names[index])
             vmxml_backup = vmxml.copy()
             vmxml_backups.append(vmxml_backup)
-            huge_pages_num += 1024
+            if int(vmxml.max_mem) < 1024000:
```
Comment (dzhengfy) [inline libvirt/tests/src/virtual_device/filesystem_device.py:145]: vmxml.max_mem is already an integer. No need int()
And I am not sure if vmxml.max_mem is None or not when there is no <maxMemory> setting in the guest xml. If possible None, we have to judge it

Comment (yafu-1) [reply libvirt/tests/src/virtual_device/filesystem_device.py:145]: vmxml.max_mem is the value of ' <memory unit='KiB'>2097152</memory> ' and this setting can not lack in the guest xml.

**Pull request merged**